lib/web_ui/lib/src/engine/text/word_break_properties.dart

```diff
@@ -46,976 +36,976 @@ enum CharProperty {
-  UnicodeRange<CharProperty>(0x1EE00, 0x1EE03, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE05, 0x1EE1F, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE21, 0x1EE22, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE24, 0x1EE24, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE27, 0x1EE27, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE29, 0x1EE32, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE34, 0x1EE37, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE39, 0x1EE39, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE3B, 0x1EE3B, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE42, 0x1EE42, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE47, 0x1EE47, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE49, 0x1EE49, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE4B, 0x1EE4B, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE4D, 0x1EE4F, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE51, 0x1EE52, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE54, 0x1EE54, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE57, 0x1EE57, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE59, 0x1EE59, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE5B, 0x1EE5B, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE5D, 0x1EE5D, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE5F, 0x1EE5F, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE61, 0x1EE62, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE64, 0x1EE64, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE67, 0x1EE6A, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE6C, 0x1EE72, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE74, 0x1EE77, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE79, 0x1EE7C, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE7E, 0x1EE7E, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE80, 0x1EE89, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EE8B, 0x1EE9B, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EEA1, 0x1EEA3, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EEA5, 0x1EEA9, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1EEAB, 0x1EEBB, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1F130, 0x1F149, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1F150, 0x1F169, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1F170, 0x1F189, CharProperty.ALetter),
-  UnicodeRange<CharProperty>(0x1F1E6, 0x1F1FF, CharProperty.RegionalIndicator),
-  UnicodeRange<CharProperty>(0x1F3FB, 0x1F3FF, CharProperty.Extend),
-  UnicodeRange<CharProperty>(0xE0001, 0xE0001, CharProperty.Format),
-  UnicodeRange<CharProperty>(0xE0020, 0xE007F, CharProperty.Extend),
-  UnicodeRange<CharProperty>(0xE0100, 0xE01EF, CharProperty.Extend)
+const UnicodePropertyLookup<WordCharProperty> wordLookup =
```

Comment:
These can be encoded into a single string using fixed-width hex + 1-character code for `WordCharProperty`. If the entire range is fixed width you can get rid of `0x` and comma delimiters. For example, this:

```dart
UnicodeRange<WordCharProperty>(0x000A, 0x000A, WordCharProperty.LF),
UnicodeRange<WordCharProperty>(0x000B, 0x000C, WordCharProperty.Newline),
UnicodeRange<WordCharProperty>(0x000D, 0x000D, WordCharProperty.CR),
```

Contains the following information:

```
0x000A, 0x000A, WordCharProperty.LF
0x000B, 0x000C, WordCharProperty.Newline
0x000D, 0x000D, WordCharProperty.CR
```

Let's use 5 hex digits for every code, and use 'A' for `LF`, 'B' for `Newline`, 'C' for `CR`. We get:

```
0x0000A, 0x0000A, A
0x0000B, 0x0000C, B
0x0000D, 0x0000D, C
```

Now everything is fixed-width, so we can drop the `0x` and delimiters:

```
0000A0000AA0000B0000CB0000D0000DC
<   ><   >|<   ><   >|<   ><   >|
```

Reply:
Thanks for the suggestion! This gave me the idea of using a base36 encoding which leads to a maximum of 4 characters for all the integers in our range. That, combined with encoding each enum value as a letter (I had to make it `a-z` and `A-Z` case-sensitive because there are 43 enum values) produced really good results.

I'll be working on unpacking now and I'll share the performance of that once I have it.